src/lib/input/input-container.ts

```diff
@@ -311,9 +295,8 @@ export class MdInputDirective implements OnChanges, OnDestroy, DoCheck {
   /** Re-evaluates the error state. This is only relevant with @angular/forms. */
   private _updateErrorState() {
     const oldState = this._isErrorState;
-    const control = this._ngControl;
-    const parent = this._parentFormGroup || this._parentForm;
-    const newState = control && this.errorStateMatcher(control.control as FormControl, parent);
+    const errorMatcher = this.errorStateMatcher || this._errorOptions.errorStateMatcher;
```

Comment:
I'm not sure how making the `ErrorStateMatcher` would help, considering that it would still look something like:

```
class ErrorStateMatcher {
  match(): boolean;
}
```

Otherwise I'll fix it so it doesn't lose the context.